File: 2032-two-out-of-three/2032-two-out-of-three.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @param {number[]} nums3
 * @return {number[]}
 */
var twoOutOfThree = function(nums1, nums2, nums3) {
    let arr = [...new Set(nums1), ...new Set(nums2), ...new Set(nums3)];
    let map = {};
    let results = [];
    
    for(let num of arr) {
        map[num] = (map[num] || 0) + 1;
    }
    
    for(key in map) {
        if(map[key] >= 2) {
            results.push(key);
        }
    }
    return results;
};

File: 2032-two-out-of-three/2032-two-out-of-three.py
class Solution:
    def twoOutOfThree(self, nums1: List[int], nums2: List[int], nums3: List[int]) -> List[int]:
        set1 = set(nums1)
        set2 = set(nums2)
        set3 = set(nums3)
        common_nums = set1.intersection(set2) | set1.intersection(set3) | set2.intersection(set3)
        results = list(common_nums)
        return results